src/ui/screens/Parachains/CrowdloanScreen.tsx

```diff
@@ -69,6 +69,7 @@ export function CrowdloanScreen({navigation}: ScreenProps) {
                 active={key === ongoingKey}
                 onPressContribute={() => {
                   setOpenContributeId(item.paraId);
+                  refetchCrowdloans();
```

Comment:
When this item is pressed, no Tx is being sent.. This just opens the modal.

You need to fetch the query in the `ContributeBox` component. You can use [useLazyQuery](https://www.apollographql.com/docs/react/data/queries#manual-execution-with-uselazyquery) to fetch only after the Tx has resolved.

Reply:
The `useCrowdloan` hook will get heavy if we start to add `useLazyQuery,` Thanks to @ishrak-imam  for helping me that I can pass refetch as a prop and invoke the query.

Reply:
What do you mean by heavy?

Reply:
The purpose of the hook is just to fetch the active and ended, while adding lazyQuery will make it broader, Thereby we will have to create a new hook for lazyQuery, and since our case, we are rendering the same list, it would be additional work to invoke the lazyQuery client and assign the values back to the `useCrowdloan` variables since they are already declared as constants.

Reply:
The idea is to expose another hook like `useLazyAllCrowdloans` or `useFetchAllCrowdloans` with `useLazyQuery` that makes use of the same query internally, you don't have to assign values or do anything else, the query will be cached as well